api/engine.py

```diff
@@ -318,6 +324,77 @@ def __get_result_content():
                                    msg="your scan id is not valid!")), 400
     return __get_result(__language(), id)
 
+@app.route("/results/get_json", methods=["GET"])
+def __get_results_json():
+    """
+    get host's logs through the API in JSON type
+
+    Returns:
+        an array with JSON events
+    """
+    session = create_connection(__language())
+    __api_key_check(app, flask_request, __language())
+    try:
+        id = __get_value(flask_request, "id")
+    except:
+        id = ""
+    result_id = session.query(Report).join(HostsLog, Report.id==HostsLog.id).filter(Report.id==id).all()
+    json_object = {}
+    if(result_id):
+        scan_id = result_id[0].scan_id
+        data = session.query(HostsLog).filter(HostsLog.scan_id==scan_id).all()
+        host = data[0].host
+        data = __logs_to_report_json(host, __language())
+        json_object = json.dumps(data)
+    return Response(json_object,
+        mimetype='application/json',
+        headers={'Content-Disposition':'attachment;filename=results.json'})
+
+@app.route("/results/get_csv", methods=["GET"])
+def __get_results_csv():
+    """
+    get host's logs through the API in JSON type
+
+    Returns:
+        an array with JSON events
+    """
+    session = create_connection(__language())
+    __api_key_check(app, flask_request, __language())
+    try:
+        id = __get_value(flask_request, "id")
+        scan_id_temp = session.query(Report).filter(Report.id==id).all()
+    except:
+        id = ""
+    if(scan_id_temp):
+        result_id = session.query(Report).join(HostsLog, Report.scan_id==HostsLog.scan_id).filter(Report.scan_id==scan_id_temp[0].scan_id).all()
+    else:
+        result_id = []
+    s = ''
+    if(result_id):
+        scan_id = result_id[0].scan_id
+        data = session.query(HostsLog).filter(HostsLog.scan_id==scan_id).all()
+        host = data[0].host
```

Comment:
Because now we have the result_id (the Object), so every member of it will be having the same host. As we now have extracted the details of that particular id from /results/get_csv?id=112. So we can take host from any member of the object. I used the very first one.